tests/ga/test_extension.py

```diff
@@ -2322,324 +2321,320 @@ def test_it_should_parse_boolean_values_encoded_as_strings(self):
+            "DisableAction's return code should be in updateAction's env.")
+
+
+    def test_uninstall_failed_env_variable_should_set_for_install_when_continue_on_update_failure_is_true(self):
+        exit_code, uninstall_action = Actions.generate_unique_fail()
+
+        first_ext = extension_emulator(uninstall_action=uninstall_action)
+        second_ext = extension_emulator(version="1.1.0", continue_on_update_failure=True)
+
+        invocation_record = TestExtensionUpdateOnFailure._do_upgrade_scenario_and_get_order(first_ext, second_ext)
+
+        invocation_record.compare(
+            (first_ext, ExtensionCommandName.DISABLE),
+            (second_ext, ExtensionCommandName.UPDATE),
+            (first_ext, ExtensionCommandName.UNINSTALL),
+            (second_ext, ExtensionCommandName.INSTALL),
+            (second_ext, ExtensionCommandName.ENABLE)
+        )
+
+        _, kwargs = second_ext.actions[ExtensionCommandName.INSTALL].call_args
+
+        self.assertEqual(kwargs["env"][ExtCommandEnvVariable.UninstallReturnCode], exit_code,
+            "UninstallAction's return code should be in updateAction's env.")
 
-            self.assertTrue('-install' in kwargs['command'] and ExtCommandEnvVariable.UninstallReturnCode in kwargs['env'],
-                            "The install command should have Uninstall Failed in env variable")
 
     def test_extension_error_should_be_raised_when_continue_on_update_failure_is_false_on_disable_failure(self, *args):
-        old_handler_i = self._get_ext_handler_instance('foo', '1.0.0')
-        old_handler_i.set_handler_state(ExtHandlerState.Enabled)
-        new_handler_i = self._get_ext_handler_instance('foo', '1.0.1', continue_on_update_failure=False)
+        exit_code, disable_action = Actions.generate_unique_fail()
 
-        with patch.object(ExtHandlerInstance, "disable", side_effect=ExtensionError("Disable Failed")):
-            with self.assertRaises(ExtensionUpdateError) as error:
-                # Ensure the error is of type ExtensionUpdateError
-                ExtHandlersHandler._update_extension_handler_and_return_if_failed(old_handler_i, new_handler_i)
+        first_ext = extension_emulator(disable_action=disable_action)
+        second_ext = extension_emulator(version="1.1.0", continue_on_update_failure=False)
 
-            msg = str(error.exception)
-            self.assertIn("Disable Failed", msg, "Update should fail with Disable Failed error")
-            self.assertIn("ExtensionError", msg, "The Exception should initially be propagated as ExtensionError")
+        invocation_record = TestExtensionUpdateOnFailure._do_upgrade_scenario_and_get_order(first_ext, second_ext)
+
+        invocation_record.compare(
+            (first_ext, ExtensionCommandName.DISABLE)
+        )
+
+        self.assertEqual(len(second_ext.status_blobs), 1, "The second extension should have a single submitted status.")
```

Comment:
I'm pretty sure you're right about that.
I don't know if that behavior falls into the scope of this particular test, but I don't really have any strong feelings about it; it would be easy enough to add, anyway, if that's what we want to do.

Reply:
If its easy to add, might as well add it. Its always good to have more testing rather than less testing

Reply:
Bump, what was the decision?
The current tests don't have this check, so I'm okay if you don't want to add it in this PR.

Reply:
This check was added! The first extension does actually report for `install`/`enable` command success, but doesn't afterward. That's why we expect a single status.